src/gui/render.cpp

```diff
@@ -626,6 +632,21 @@ bool RENDER_MaybeAutoSwitchShader([[maybe_unused]] const uint16_t canvas_width,
 #endif // C_OPENGL
 }
 
+void RENDER_NotifyEgaModeWithVgaPalette()
+{
+	auto video_mode = VGA_GetCurrentVideoMode();
```

Comment:
Yeah, good idea. It's kinda protected already, but on my first attempt it wasn't so i was getting random crashes in Hercules mode because it started doing _something_ in that mode too 😅